tcms/testruns/models.py

```diff
@@ -127,12 +127,12 @@ def belong_to(self, user):
 
         return False
 
-    def check_all_case_runs(self, case_run_id=None):
-        tcrs = self.case_run.all()
-        tcrs = tcrs.select_related('case_run_status')
+    def check_all_case_runs(self):
+        test_case_runs = self.case_run.all()
+        test_case_runs = test_case_runs.select_related('case_run_status')
 
-        for tcr in tcrs:
-            if not tcr.is_finished():
+        for test_case_run in test_case_runs:
+            if not test_case_run.is_finished():
```

Comment:
A few notes:

1) this method is not used so remove it.

2) As it is the select_related() above is just pointless because it is not used inside the loop.

3) This loop can be optimized to use a single query, see what `is_finished()` does. You can filter/exclude based on the status name and use .count() or .exists() to return a boolean.

Btw `is_finished` from both TestCaseRunStatus and TestCaseRun also becomes unused when you remove check_all_case_runs so delete them as well.